apps/src/StudioApp.js

```diff
@@ -608,6 +608,61 @@ StudioApp.prototype.init = function (config) {
   }
 };
 
+StudioApp.prototype.hasValidContainedLevelResult_ = function () {
+  var results = this.getContainedLevelResults();
+  if (results.length !== 1) {
+    throw "Exactly one contained level result is currently required.";
+  }
+  var firstResult = results[0];
+  return typeof firstResult.result.response !== 'undefined' &&
+      firstResult.result.response !== '';
+};
+
+StudioApp.prototype.notifyInitialRenderComplete = function (config) {
+  if (config.hasContainedLevels && config.containedLevelOps) {
+    this.lockContainedLevelAnswers = config.containedLevelOps.lockAnswers;
+    this.getContainedLevelResults = config.containedLevelOps.getResults;
+
+    $('#containedLevel0').appendTo($('#containedLevelContainer'));
+
+    if (this.hasValidContainedLevelResult_()) {
+      // We already have an answer, don't allow it to be changed, but allow Run
+      // to be pressed so the code can be run again.
+      this.lockContainedLevelAnswers();
+    } else {
+      // No answers yet, disable Run button until there is an answer
+      $('#runButton').prop('disabled', true);
+
+      config.containedLevelOps.registerAnswerChangedFn(function (levelId) {
```

Comment:
Now that we support ES6 :tada: this is a perfect place for an arrow function (which, note, implicitly binds `this`):

```javascript
config.containedLevelOps.registerAnswerChangedFn(levelId => {
  $('#runButton').prop('disabled', !this.hasValidContainedLevelResult_());
});
```